### PR TITLE
Removed lazy instantiation of containers

### PR DESCRIPTION
**Pull request by d-michail**
Follow up of #500 as discussed in #520 to disable lazy instantiation of vertex containers. Also contains a minor type warnings cleanup.

### CODE REVIEW

**Comment by d-michail** [comment]
@jsichi @Yimismi In a completely unrelated build, I show a failed test like this

```
Running org.jgrapht.graph.concurrent.AsSynchronizedGraphTest
....................
Time: 0.007

OK (20 tests)

....
Time: 0.001

OK (4 tests)

...
Time: 0.008

OK (3 tests)

.....
Time: 0.004

OK (5 tests)

....................
Time: 0.005

OK (20 tests)

Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.087 sec <<< FAILURE!
```
and then later on

```
Results :

Failed tests:   testScenario(org.jgrapht.graph.concurrent.AsSynchronizedGraphTest)

Tests run: 1061, Failures: 1, Errors: 0, Skipped: 0
```

It is possible this is related to the "lazy" initialization, or is there some other edge case?

File: jgrapht-core/src/test/java/org/jgrapht/graph/concurrent/AsSynchronizedGraphTest.java [hunk 1]
```diff
@@ -338,10 +344,12 @@ public void execute()
             g.setCache(!g.isCacheEnabled());
```
Comment (Yimismi) [inline jgrapht-core/src/test/java/org/jgrapht/graph/concurrent/AsSynchronizedGraphTest.java]: So sorry for the bug. This method should be synchronized because two visits of the graph may be interrupted.

**Comment by jsichi** [comment]
Thanks!  I overlooked the generics during review.

After merge, as part of the credits, I'll remove the workaround from AsSynchronizedGraph, and also add the synchronized in the test per @Yimismi.  If you get another failure after pulling that, let me know since there could be other race conditions lurking.
